server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -176,6 +176,16 @@ public enum XContentContext {
     public interface Custom extends NamedDiffable<Custom>, ToXContentFragment, ClusterState.FeatureAware {
 
         EnumSet<XContentContext> context();
+
+        static Custom fromXContent(XContentParser parser, String name) throws IOException {
+            // handling any Exception is caller's responsibility
+            return parser.namedObject(Custom.class, name, null);
+        }
+
+        static Custom fromXContent(XContentParser parser) throws IOException {
```

Comment:
@shiv0408 why breaking changes check fails


There are existing classes with existing static method signatures, that implement `Metadata.Custom`, just one of the examples:
```
public class ComposableIndexTemplateMetadata implements Metadata.Custom {
    ....
    public static ComposableIndexTemplateMetadata fromXContent(XContentParser parser) throws IOException {
        return PARSER.parse(parser, null);
    }
}
```

Now you are trying to add the conflicting static methods to the `Metadata.Custom`, changing the existing static method signatures.

Reply:
Thanks @reta for highlighting this. This method was added to be able to access fromXContent from the interface itself. As it is conflicting with other concrete classes method, needs to be fixed before attempting to backport.

Reply:
@reta @shwetathareja I agree that this method is a little confusing, as this method was not getting used anywhere I have removed this for now. But the breaking changes check is still failing due to the other method that is added above, which is not conflicting with any method in implementations of this interface.

Reply:
If we merge this backport PR, will raise a separate PR to remove the method and backport that to 2.x.

Reply:
@reta  
```
        static Custom fromXContent(XContentParser parser, String name) throws IOException {
            // handling any Exception is caller's responsibility
            return parser.namedObject(Custom.class, name, null);
        }
```
This has default implementation and shouldn't be considered as breaking change for plugins right?

Reply:
@andrross  ```static Custom fromXContent(XContentParser parser, String name)``` this shouldn't be breaking change as it has default implementation.

Reply:
@andrross the method ```static Custom fromXContent(XContentParser parser)``` which was causing conflict is already removed via https://github.com/opensearch-project/OpenSearch/pull/13869 . I am not sure if we merge this, all future PRs would fail. For the future PRs, this should be baseline. We need to unblock this backport.

Reply:
The baseline is the released 2.14 artifact. If we merge this then I believe the compatibility check will fail for all 2.x PRs as the code on this branch will be flagged as incompatible with 2.14. The options to unblock here that I see are:

- Work around the bug by moving the static method `fromXContent(XContentParser, String)` from the interface to somewhere else (such as in the Metadata class).
- Disable the compatibility check
- Figure out how to get the latest version of the japicmp library to be used by our github action, which should fix the false positive here

Reply:
@shwetathareja To get unblocked, could you try to update the version of `siom79/japicmp` and then re-run the compability check command.  If that works, then lets include the version bump in this PR and forward port to main?

Reply:
@peternied The issue is the [gradle plugin](https://plugins.gradle.org/plugin/me.champeau.gradle.japicmp) we're using has not been updated to pull in the latest version of the underlying japicmp library.

Reply:
I've created an issue on japicmp-gradle-plugin to release a new version [1], it has the updated dependency but hasn't released.  To workaround add a `implementation` dependency in gradle for the updated version of japicmp. 

- [1] https://github.com/melix/japicmp-gradle-plugin/issues/72

Reply:
@andrross I wouldnt prefer making a code change just to override japicmp issue.

Reply:
@peternied I tried using the `implementation` dependency but it the command was still failing, possibly I was not overriding it correctly? But I tried publishing the jar to maven local and use that in our build and check was passing in that case. Can we somehow use that jar in our build while running this check?